libraries/eosiolib/contracts/eosio/key_value.hpp

```diff
@@ -880,11 +881,15 @@ class kv_table : kv_detail::kv_table_base {
     * If the put attempts to store over an existing secondary index, the transaction will be aborted.
     *
     * @param value - The entry to be stored in the table.
+    * @param payer - The payer for the entry.
     */
-   void put(const T& value) {
-      uint32_t value_size;
+   void put(const T& value, eosio::name payer) {
       T old_value;
-      kv_table_base::put(&value, &old_value, &get_size_fun, &deserialize_fun, &serialize_fun);
+      kv_table_base::put(&value, &old_value, &get_size_fun, &deserialize_fun, &serialize_fun, payer);
+   }
+
+   void put(const T& value) {
+      put(value, kv_table_base::contract_name);
```

Comment:
My preference would be to have KV mirror multi-index and not provide a default like this and instead always require the payer to be specified. The contract can use `get_self()` to pay for the resources if they choose.